source/dub/packagesuppliers/fallback.d

```diff
@@ -40,19 +39,33 @@ private template fallback(T, alias func)
 {
 	import std.format : format;
 	enum fallback = q{
-		import std.range : back, dropBackOne;
-		import dub.internal.vibecompat.core.log : logDebug;
-		scope (failure)
+		import dub.internal.vibecompat.core.log : logDiagnostic;
+
+		Exception firstEx;
+		try
+			return m_suppliers[0].%1$s(args);
+		catch (Exception e)
+		{
+			logDiagnostic("Package supplier %%s failed with '%%s', trying fallbacks.",
+				m_suppliers[0].description, e.msg);
+			m_suppliers = m_suppliers[1 .. $];
```

Comment:
Not sure whether this is a good idea when dub is used as a library and potentially runs for days, but I think only DLS does so and I'm just sending a quick ping to @LaurentTreguier to make him aware of this change.